python/ray/data/_internal/execution/operators/hash_shuffle.py

```diff
@@ -1096,6 +1100,18 @@ def _estimate_aggregator_memory_allocation(
         return aggregator_total_memory_required
 
 
+@dataclass
+class AggregatorHealthInfo:
+    """Health information about aggregators for issue detection."""
+
+    started_at: float
+    ready_aggregators: int
+    total_aggregators: int
+    has_unready_aggregators: bool
+    wait_time: float
+    required_cpus: float
```

Comment:
Let's make this required resources, since aggregators also provision memory (and make sure we handle that as well)